silkworm/silkrpc/json/block.cpp

```diff
@@ -164,20 +164,43 @@ struct GlazeJsonNullBlockReply {
     };
 };
 
-void make_glaze_json_null_content(uint32_t id, std::string& json_reply) {
+void make_glaze_json_null_content(const nlohmann::json& request_json, std::string& json_reply) {
     GlazeJsonNullBlockReply block_json_data{};
-    block_json_data.id = id;
+    if (request_json.contains("id")) {
+        const auto& id = request_json["id"];
+        if (id.is_number()) {
+            block_json_data.id = id.get<std::uint32_t>();
+        } else if (id.is_string()) {
+            block_json_data.id = std::make_shared<std::string>(id.get<std::string>());
+        } else {
+            block_json_data.id = nullptr;
+        }
+    } else {
+        block_json_data.id = nullptr;
+    }
```

Comment:
I suggest to extract this piece of code (which is repeated in multiple places) into a dedicated method `make_jsonrpc_id` with the following signature:

`JsonRpcId make_jsonrpc_id(const nlohmann::json& request_json)`